### PR TITLE
Dreaming current fix(?) and tweak

### PR DESCRIPTION
**Pull request by EgorDinamit**
## About The Pull Request

- Added max distance variable to dreaming current and therefore increase it from default 30 to 40.
- Added a simple loop to finding the path to next node, so it doesn't fail all that often.
- DC won't play sound and animation if there's nowhere to move.

## Why It's Good For The Game

- Fixes an issue DC had on big maps where it would circle through same department forever.
- A fail-safe, nothing else.
- Fix. DC doesn't look dumb anymore.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/abnormality/waw/dreaming_current.dm [hunk 1]
```diff
@@ -103,10 +114,20 @@
 		var/turf/T = get_closest_atom(/turf/open, potential_turfs, picking_from)
 		if(!T)
 			break
-		movement_path += get_path_to(path_start, T, /turf/proc/Distance_cardinal)
+		var/list/our_path = list()
+		for(var/o = 1 to 3) // Grand total of 3 retries
+			our_path = get_path_to(path_start, T, /turf/proc/Distance_cardinal, dash_max_distance)
+			if(islist(our_path) && LAZYLEN(our_path))
+				break
+			potential_turfs -= T // Couldn't find path to it, don't try again
```
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/dreaming_current.dm:122]: Wouldn't this result in a duplicate removal if it initially could not get a path, but later managed to on, say loop 2 or 3?

**Review by dsmith328 — COMMENTED** [review]
Seems good, would like comment addressed before strict approval is given.

File: code/modules/mob/living/simple_animal/abnormality/waw/dreaming_current.dm (hunk 1 above)
Comment (EgorDinamit) [reply code/modules/mob/living/simple_animal/abnormality/waw/dreaming_current.dm:122]: Right, I will have to pick a new T variable here.

**Review by dsmith328 — APPROVED** [review]
More range and work exclusion are both very good changes for him

**Pull request merged**